### DISCUSSION TITLE
Magnetometer standard deviation

### DISCUSSION THREAD

**Discussion by BrunoOtavio98** [rodralez/NaveGo#59] [Q&A] [answered]
In ins_gnss_mag.m (line 46), the magnetometer standard deviation is specified in rad, but didn't it have to be in uT or G?

**Comment by rodralez** [comment, marked as answer]
Hello Bruno,

Good question. 

The magnetometer standard deviation is in radians because the Kalman filter tries to estimate the yaw error ![\delta\psi](https://latex.codecogs.com/svg.latex?\space\delta\psi) using the difference between the magnetometer compass ![\psi_{mag}](https://latex.codecogs.com/svg.latex?\space\psi_{mag}) and the INS yaw angle  ![\psi}](https://latex.codecogs.com/svg.latex?\space\psi) (ins_gnss_mag.m, line 353).

The Kalman filter is not configured to denoise the magnetometer measurements. 

According to my short experience, the yaw angle estimation (nav_e.yaw) is not improved dramatically with this Kalman filter scheme. 

It would be nice if other NaveGo users share their experiences using magnetometers in integrated navigation systems.

Best regards.
